habitat_baselines/il/trainers/pacman_trainer.py

```diff
@@ -0,0 +1,663 @@
+                                elif action == 2:
+                                    my_action = 3  # right
+                                elif action == 3:
+                                    my_action = 0  # stop
+
+                            observations = env.sim.step(my_action)
+                            img = observations["rgb"]
+                            first_step = False
+
+                        # compute stats
+                        m = "" if j == "pred" else "_f"
+                        metrics_slug[
+                            "d_T{}_{}".format(m, i)
+                        ] = dists_to_target[-1]
+                        metrics_slug["d_D{}_{}".format(m, i)] = (
+                            dists_to_target[0] - dists_to_target[-1]
+                        )
+                        metrics_slug["d_min{}_{}".format(m, i)] = np.array(
+                            dists_to_target
+                        ).min()
+
+                        if j != "fwd-only":
+                            metrics_slug[
+                                "ep_len_{}".format(i)
+                            ] = episode_length
+                            if action == 3:
+                                metrics_slug["stop_{}".format(i)] = 1
+                            else:
+                                metrics_slug["stop_{}".format(i)] = 0
+
+                            if i != "rand_init":
+                                metrics_slug[
+                                    "d_0_{}".format(i)
+                                ] = dists_to_target[0]
+
+                # collate and update metrics
+                metrics_list = []
+                for ind, i in enumerate(metrics.metric_names):
+                    if i not in metrics_slug:
+                        metrics_list.append(metrics.metrics[ind][0])
+                    else:
+                        metrics_list.append(metrics_slug[i])
+
+                # update metrics
+                metrics.update(metrics_list)
+
+                if t % config.LOG_INTERVAL == 0:
+                    logger.info(
+                        "Valid cases: {}; Invalid cases: {}".format(
+                            (t + 1) * 4 - len(invalids), len(invalids)
+                        )
+                    )
+                    logger.info(
+                        "EVAL: Avg metrics: {}".format(
+                            metrics.get_stat_string(mode=0)
+                        )
+                    )
+                    print(
+                        "-----------------------------------------------------"
+                    )
+
+                if (
+                    config.EVAL_SAVE_RESULTS
+                    and t % config.EVAL_SAVE_RESULTS_INTERVAL == 0
+                ):
+                    q_string = q_vocab_dict.token_idx_2_string(question[0])
+                    logger.info("Question: {}".format(q_string))
+
+                    self._save_nav_results(
```

Comment:
This should probably follow the `VIDEO` option and only write if that is set to `DISK`.

Reply:
Added in latest commit – marking as resolved.